pkg/restapi/rp/operation/operations.go

```diff
@@ -168,22 +195,39 @@ func (o *Operation) hydraLoginHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (o *Operation) setLoginRequestForState(state string, request *models.LoginRequest) {
-	o.lock.Lock()
-	defer o.lock.Unlock()
+	o.oidcStateLock.Lock()
```

Comment:
How does this impact running a cluster of adapters and also adapter restarts?

Reply:
@troyronda 

> How does this impact running a cluster of adapters and also adapter restarts?

This implementation does not survive either. Good catch.

We would have to implement sticky transient storage to resume after restarts. We mount that at the cluster level to avoid reliance on sticky sessions.

Reply:
I've modified the description of #29 to account for this.

Reply:
@troyronda I'm merging this PR, will followup on your point in #29